### PR TITLE
crossdock: Add an apachethrift crossdock behavior

### PR DESCRIPTION
**Pull request by abhinav**
This adds a crossdock behavior that tests that YARPC clients can call Apache
Thrift HTTP servers. This tests both, multiplexed and non-multiplexed Thrift
servers.

### CODE REVIEW

File: crossdock/client/apachethrift/behavior.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		{
+			ServerName: "Multiplexed",
+			Services:   gauntlet.AllServices,
+			Options:    []thrift.ClientOption{thrift.Multiplexed},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Tag("outbound", tt.ServerName)
+		gauntlet.RunGauntlet(t, gauntlet.Config{
```
Comment (breerly) [inline crossdock/client/apachethrift/behavior.go]: wow, reusing the `guantlet.RunGauntlet` again! freaking yes!!!!

File: crossdock/main_test.go [hunk 2]
```diff
@@ -109,6 +109,13 @@ func TestCrossdock(t *testing.T) {
 				"ctxclient": "127.0.0.1",
 			},
 		},
+		{
+			name: "apachethrift",
```
Comment (breerly) [inline crossdock/main_test.go]: thanks for not forgetting this

File: crossdock/server/http/apache.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package http
```
Comment (breerly) [inline crossdock/server/http/apache.go]: This server should live in `crossdock/server/apachethrift/server.go` - let's not conflate these 2 servers.

File: crossdock/server/http/server.go [hunk 4]
```diff
@@ -20,66 +20,14 @@
...
-	}
-	procedure := req.Header.Get(`RPC-Procedure`)
-	if f, ok := m.handlers[procedure]; ok {
-		f.ServeHTTP(w, req)
-	} else {
-		w.WriteHeader(http.StatusBadRequest)
-		fmt.Fprintf(w, "Unknown procedure: %q\n", procedure)
-	}
+	startThirdParty()
+	startApacheThrift()
```
Comment (breerly) [inline crossdock/server/http/server.go]: I really don't want these both in the same package.

File: docker-compose.yml [hunk 5]
```diff
@@ -42,13 +44,14 @@ services:
             - BEHAVIOR_TIMEOUT=client,server,transport
             # BEHAVIOR_INBOUNDTTL TODO
             - BEHAVIOR_CTXPROPAGATION=ctxclient,ctxserver,transport
-            - REPORT=compact
+            - BEHAVIOR_APACHETHRIFT=apachethriftclient,apachethriftserver
+            - REPORT=list
```
Comment (breerly) [inline docker-compose.yml]: Was this intentional?

File: crossdock/server/http/apache.go (hunk 3 above)
Comment (abhinav) [inline crossdock/server/http/apache.go]: 👍

File: crossdock/server/http/server.go (hunk 4 above)
Comment (abhinav) [inline crossdock/server/http/server.go]: 👍

File: docker-compose.yml (hunk 5 above)
Comment (abhinav) [inline docker-compose.yml]: derp

**Pull request merged**